ONSdigital/statistical-methods-library
# Estimation spp-3031

- Starting Estimation.
- Maybe right. By Expansion at cell level return.
- More Maths.
- Combine estimation into a single function
- Fix handling of defaults for death marker and h value and incorrect aliasing in returned data frame in estimation
- Add docstring to estimation
- Fix documentation formatting and add comments
- Fix capitalisation in a comment in estimation
- Rename parameters such that their aliased names are the same but just without the _col suffix and add not null check as per documentation in estimation.
- Add validation as per the documentation and improve comments
- Rename estimation to estimate in estimation module
- Basic Test Data Input.
- Restructure tests directory in preparation for adding estimation tests
- Reformat with black and isort
- Some Data
- Moved Data.
- selectExpr after groupBy is not a thing, fix this code in estimation
- Start Of Tests.
- Wrong Type.
- Black.
- Show Me
- And Not OR.
- Brackets.
- Attempt to work arround aliases not working in the same select statement
- Add h_value to expansion estimation design weight select
- We have to join h value back on after the group by in expansion estimation
- No need to join on h value, just grab the first one per period and strata in the aggregation in expansion estimation
- Correct join in the case of Combined Ratio estimation
- Test Adjustment
- Missing Column.
- Make estimation return the correct columns depending on the estimation being performed and document this
- Super Basic Data For Estimation.
- Renamed.
- Schema.
- Schema.
- Make checks for auxiliary and calibration group data check if the column names provided are None as this is a bit more explicit
- death marker is an int so set the literal appropriately when none is provided
- Add automated test running logic in estimation tests similar to imputation tests
- Reformat estimation tests with black and isort
- Fix variable naming to prevent variable shadowing in estimation tests
- Fix variable names to prevent shadowing in imputation tests
- Added decorator to limit type checking and tidied line spacings
- Added unit tests for validation plus basic unit test data.
- Added unit tests for further validation.
- Added unit tests for further validation.
- Added unit tests for further validation. Tidied.
- Extra Data
- estimation test data non_boolean_marker -> non_boolean_markers
- make params a tuple in estimation tests
- Fix checking unspecified death marker column in estimation
- Baby Black.
- Corrected test data for non-bool sample marker.
- reformat imputation tests with black
- Tidied test comments, please ignore.
- Documentation. (*sigh*)
- The Basic Doc.
- Documentation continued.
- Data From Methodology.
- Required Inputs.
- Odd Data
- Test Data Columns Names Need To Match.
- Test Data Columns Names Need To Match.
- Mistake In Calculation.
- Decrease Precision.
- Style.
- Update Docs.
- Documentation continued.
- Renamed Methodology Scenarios.
- Tidying test file names.
